src/Nethermind/Nethermind.Consensus/Stateless/StatelessBlockProcessingEnv.cs

```diff
@@ -0,0 +1,104 @@
+using Nethermind.Core;
+using Nethermind.Core.Crypto;
+using Nethermind.Core.Specs;
+using Nethermind.Db;
+using Nethermind.Evm;
+using Nethermind.Evm.State;
+using Nethermind.Evm.TransactionProcessing;
+using Nethermind.Logging;
+using Nethermind.State;
+using Nethermind.Trie;
+using Nethermind.Trie.Pruning;
+
+namespace Nethermind.Consensus.Stateless;
+
+public class StatelessBlockProcessingEnv(
+    Witness witness,
+    ISpecProvider specProvider,
+    ISealValidator sealValidator,
+    ILogManager logManager)
+{
+    private IBlockProcessor? _blockProcessor;
+    public IBlockProcessor BlockProcessor
+    {
+        get => _blockProcessor ??= GetProcessor();
+    }
+
+    private IWorldState? _worldState;
+    public IWorldState WorldState
+    {
+        get => _worldState ??= new WorldState(CreateTrie(), CreateCodeDb(), logManager);
+    }
+
+    private IBlockProcessor GetProcessor()
+    {
+        IBlockTree statelessBlockTree = new StatelessBlockTree(witness.DecodedHeaders);
+        ITransactionProcessor txProcessor = CreateTransactionProcessor(WorldState, statelessBlockTree);
+        IBlockProcessor.IBlockTransactionsExecutor txExecutor =
+            new BlockProcessor.BlockValidationTransactionsExecutor(
+                new ExecuteTransactionProcessorAdapter(txProcessor),
+                WorldState);
+
+        IHeaderValidator headerValidator = new HeaderValidator(statelessBlockTree, sealValidator, specProvider, logManager);
+        IBlockValidator blockValidator = new BlockValidator(new TxValidator(specProvider.ChainId), headerValidator,
+            new UnclesValidator(statelessBlockTree, headerValidator, logManager), specProvider, logManager);
+
+        return new BlockProcessor(
+            specProvider,
+            blockValidator,
+            NoBlockRewards.Instance,
+            txExecutor,
+            WorldState,
+            NullReceiptStorage.Instance,
+            new BeaconBlockRootHandler(txProcessor, WorldState),
+            new BlockhashStore(specProvider, WorldState),
+            logManager,
+            new WithdrawalProcessor(WorldState, logManager),
+            new ExecutionRequestsProcessor(txProcessor)
+        );
+    }
+
+    private ITrieStore CreateTrie()
+    {
+        IKeyValueStore db = new MemDb();
+        foreach (var stateElement in witness.State)
+        {
+            var hash = ValueKeccak.Compute(stateElement).Bytes;
+            db.PutSpan(hash, stateElement);
+        }
+
+        NodeStorage nodeStorage = new(db, INodeStorage.KeyScheme.Hash);
+        return new TrieStore(nodeStorage, NoPruning.Instance, NoPersistence.Instance, new PruningConfig(), logManager);
+    }
+
+    private IKeyValueStoreWithBatching CreateCodeDb()
+    {
+        IKeyValueStoreWithBatching db = new MemDb();
+        foreach (var code in witness.Codes)
+        {
+            var hash = ValueKeccak.Compute(code).Bytes;
+            db.PutSpan(hash, code);
+        }
+        return db;
+    }
```

Comment:
maybe we can move these function to witness class?
so we can simply do witness.CreateTrie() and witness.CreateCodeDb()

Reply:
it would be beneficil in the case where we change the witness format, we wont have to change this class.
Also this way we can support multiple witness formats.